Library/Homebrew/cmd/install.rb

```diff
@@ -278,48 +278,26 @@ def install
     # formula was found, but there's a problem with its implementation).
     $stderr.puts e.backtrace if Homebrew::EnvConfig.developer?
     ofail e.message
-  rescue FormulaOrCaskUnavailableError => e
-    if e.name == "updog"
+  rescue FormulaOrCaskUnavailableError, Cask::CaskUnavailableError => e
+    # formula name or cask token
+    name = e.try(:name) || e.token
+
+    if name == "updog"
       ofail "What's updog?"
       return
     end
 
     opoo e
-    ohai "Searching for similarly named formulae..."
-    formulae_search_results = search_formulae(e.name)
-    case formulae_search_results.length
-    when 0
-      ofail "No similarly named formulae found."
-    when 1
-      puts "This similarly named formula was found:"
-      puts formulae_search_results
-      puts "To install it, run:\n  brew install #{formulae_search_results.first}"
-    else
-      puts "These similarly named formulae were found:"
-      puts Formatter.columns(formulae_search_results)
-      puts "To install one of them, run (for example):\n  brew install #{formulae_search_results.first}"
-    end
+    ohai "Searching for similarly named formulae and casks..."
 
-    if (reason = MissingFormula.reason(e.name))
-      $stderr.puts reason
-      return
-    end
+    # Don't treat formula/cask name as a regex
+    query = string_or_regex = name
+    if search_names(query, string_or_regex, args)
```

Comment:
Some of these variables aren't functionally necessary but they help improve readability.